Utils/API/server/lib/dkb/api/storages/__init__.py

```diff
@@ -77,3 +77,30 @@ def task_kwsearch(**kwargs):
     :rtype: list
     """
     return es.task_kwsearch(**kwargs)
+
+
+def task_derivation_statistics(**kwargs):
+    """ Calculate statistics of derivation efficiency.
+
+    Resulting statistics has the following structure:
+    {
+      'data': {
+        ...
+        'some_output_format_name': {
+          'total': 123,
+          'ratio': 0.456,
+          'events_ratio': 0.789
+        },
```

Comment:
'total' is a number of tasks, right?

Reply:
Yes. It is ES's term which is used in Big PanDA Monitoring and here - in the former it is replaced with "tasks" before returning the response. I've made no such replacement here because, as I've said in a comment, such matters depend on where and how this data will be handled next.

Reply:
Used in ProdTask interface, you meant to say (it is built upon the core of BigPanDA Monitor, but it is a different instance and different team (M.Borodin exclusively, if I get it correctly)).

Anyway: ES says "total", meaning "total number of matching documents" (since it doesn't know if these documents are "tasks" or what). We, on the other hand, are fully aware of the meaning of the data we are providing -- so why not make things a bit more user-friendly?

Reply:
> @Evildoor,
> The idea is that REST API method returns data in the form ProdTask expects. For now, ProdTask constructs `result` as `List` of `Hash`es (check hash format [here](https://github.com/PanDAWMS/panda-bigmon-atlas/blob/052aebea3dcb3f1a5f4e9e7249c2ffc87f2bcb92/atlas/dkb/views.py#L931-L932)), sorted by `"output"` value.
> 
> Why have you decided to return different format?

Because I mistakenly thought that the resulting format is yet to be decided, while it is decided by ProdTask expectations. My bad, changed the format.

> Also, note the `"task_ids"` key, value of which is empty in ProdTask version for now. Could you provide value for it as well?

Done.

Reply:
Update: while 8cf6a28 allows `task_ids` to be retrieved, it only gets them for 10 tasks or less, because it's the default behavior for searches in ES, to get one "page" of results. Currently, each format has less than 50 tasks, so this can be "fixed" by changing the size of the "page" from 10 to 100-500-etc.

However, in arbitrary case we don't know the amount of tasks, and to properly get all the ids of all the tasks (in request) it is necessary, if I understand correctly, to use [scrolling](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-body.html#request-body-search-scroll), which "... is not intended for real time user requests".

What should we do?

Reply:
For now we can provide accurate value for the number of tasks (which is already there in `"tasks"` field) and, say, first 100 IDs. This will allow Mikhail to test usability of this data, output some notification if there's some "hidden" data -- and see if users ever need to see more than these 100 tasks.

I believe it shouldn't significantly affect the statictics calculation time, but should you see the opposit -- we can reduce the value to 50 or even get back to default 10.

Reply:
> > > The idea is that REST API method returns data in the form ProdTask expects. For now, ProdTask constructs result as List of Hashes (check hash format [here](https://github.com/PanDAWMS/panda-bigmon-atlas/blob/052aebea3dcb3f1a5f4e9e7249c2ffc87f2bcb92/atlas/dkb/views.py#L931-L932)), sorted by "output" value.
> > 
> > 
> > changed the format.
> 
> Have you noticed the "sorted by "output" value" part (which was added due to this [line](https://github.com/PanDAWMS/panda-bigmon-atlas/blob/052aebea3dcb3f1a5f4e9e7249c2ffc87f2bcb92/atlas/dkb/views.py#L973))? I couldn't find how it was implemented...

Missed it, my bad. Updated 735bf7d.

> 
> > Update: while [8cf6a28](https://github.com/PanDAWMS/dkb/commit/8cf6a28268eaa18181d4445bec14a47beff9fa8c) allows task_ids to be retrieved, it only gets them for 10 tasks or less, because it's the default behavior for searches in ES, to get one "page" of results. Currently, each format has less than 50 tasks, so this can be "fixed" by changing the size of the "page" from 10 to 100-500-etc.
> > However, in arbitrary case we don't know the amount of tasks, and to properly get all the ids of all the tasks (in request) it is necessary, if I understand correctly, to use scrolling, which "... is not intended for real time user requests".
> > What should we do?
> 
> For now we can provide accurate value for the number of tasks (which is already there in `"tasks"` field) and, say, first 100 IDs. This will allow Mikhail to test usability of this data, output some notification if there's some "hidden" data -- and see if users ever need to see more than these 100 tasks.
> 
> I believe it shouldn't significantly affect the statictics calculation time, but should you see the opposit -- we can reduce the value to 50 or even get back to default 10.

Good. Updated 578e3ce.